linear_operator/operators/block_matrix_linear_operator.py

```diff
@@ -0,0 +1,168 @@
+from typing import List, Optional, Tuple, Union
+
+import torch
+from jaxtyping import Float
+from torch import Tensor
+
+from ._linear_operator import IndexType, LinearOperator
+from .dense_linear_operator import DenseLinearOperator
+from .zero_linear_operator import ZeroLinearOperator
+
+
+class BlockMatrixLinearOperator(LinearOperator):
+    """
+    A TxT block matrix of LinearOperators.
+
+    Idea. Represent [TN, TM] tensors by TxT blocks of NxM lazy tensors.
+
+    Implementation. A block linear operator class that can keep track of the [T, T] block structure,
+    represented as T^2 lazy tensors of the same shape. Implement matrix multiplication between block matrices as
+    the appropriate linear operators on the blocks.
+
+    :param linear_operators: A TxT nested list of linear operators representing a 2-D matrix
+    """
+
+    def __init__(self, linear_operators: List[List[LinearOperator]]) -> None:
+        assert isinstance(linear_operators, list), f"{self.__class__.__name__} expects a nested list of LinearOperators"
```

Comment:
Small nit: could we instead assert that `linear_operators` is an iterable?

Reply:
Done!